### DISCUSSION TITLE
How to skip deploy for certain commits, e.g., a commit changes only README.md

### DISCUSSION THREAD

**Discussion by xuchunyang** [vercel/vercel#5087] [Help]
According to my understand, any commit can trigger deployment, even the commit can't affect the output, such as a commit changes just README.md. I wonder if I can skip centain files for deployment. I learnt that Travis/Netlify skips test/deploy if the commit message contains `[skip ci]`. I'm asking this because I have a [project](https://github.com/xuchunyang/epkg) which generates about 8K HTML files and the deployment takes about 8 minutes.

**Comment by johnymontana** [comment]
You can set the "Ignored Build Step" in Vercel to accomplish this. For example, I have a monorepo setup and I only want Vercel to deploy when changes are commited to a certain directly (this is set as the root Vercel project directory) so I set this as the "Ignored Build Step" value:

```
git diff HEAD^ HEAD --quiet .
```

which will only trigger builds when a commit changes something in the Vercel root project directory.

[This article](https://vercel.com/support/articles/how-do-i-use-the-ignored-build-step-field-on-vercel
) has more information and examples.
